firestore.rules

```diff
@@ -293,14 +304,14 @@ service cloud.firestore {
 
       match /documents/{docId} {
         // portal-authenticated teachers can create valid documents
-        allow create: if isAuthedTeacher() && isValidDocumentCreateRequest();
+        allow create: if isValidDocumentCreateRequest();
         // teachers can only update their own documents and only if they're valid
-        allow update: if isAuthedTeacher() && isValidDocumentUpdateRequest();
+        allow update: if isValidDocumentUpdateRequest();
         // teachers can only delete their own documents
         allow delete: if isAuthedTeacher() && userIsResourceUser();
         // teachers can read their own documents or other documents in their network
-        allow read: if (isAuthed() && (resource == null || userOwnsDocument())) ||
-          (isAuthedTeacher() && (userInResourceTeachers() || resourceInTeacherNetworks() || resourceInUserClass()))
+        allow read: if (isAuthed() && (resource == null || userOwnsDocument() || resourceInUserClass())) ||
+          (isAuthedTeacher() && (userInResourceTeachers() || resourceInTeacherNetworks()))
```

Comment:
Do you mean we need an extra check in this location, or that the `resourceInTeacherNetworks()` method should always look at class networks?

Reply:
I'm looking at this again while trying to decide if it is safe to deploy these rules so we can test Portal launches.
1. `resourceInTeacherNetworks()` is used by the class and offering rules so it wouldn't be good to update that function. So it would have to be an additional check at this location.
2. In the short term we are only showing documents in the current class the sort work tab, so this check will never be reached for those queries. So I think the only time this check might be used is if a teacher is trying to work with comments on network documents. As long as we continue to include the network field on those types of metadata documents then we don't need to fix this now.
3. The access to the comments themselves ends up being handled by userCanAccessDocument. So to support comments on network documents when we no longer include the network field, we will also have to update that userCanAccessDocument method or perhaps the function that calls it `teacherCanAccessDocument` to check the class's list of networks. 

So based on that it seems like it is safe to deploy these rules without changing this. And I think it would be good to push that work into a new story and PR at this point.